pkg/plugin/plugin.go

```diff
@@ -0,0 +1,354 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/astarte-platform/astarte-go/client"
+	"github.com/astarte-platform/astarte-go/interfaces"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// Make sure SampleDatasource implements required interfaces. This is important to do
+// since otherwise we will only get a not implemented error response from plugin in
+// runtime. In this example datasource instance implements backend.QueryDataHandler,
+// backend.CheckHealthHandler, backend.StreamHandler interfaces. Plugin should not
+// implement all these interfaces - only those which are required for a particular task.
+// For example if plugin does not need streaming functionality then you are free to remove
+// methods that implement backend.StreamHandler. Implementing instancemgmt.InstanceDisposer
+// is useful to clean up resources used by previous datasource instance when a new datasource
+// instance created upon datasource settings changed.
+var (
+	_ backend.QueryDataHandler    = (*AppEngineDatasource)(nil)
+	_ backend.CheckHealthHandler  = (*AppEngineDatasource)(nil)
+	_ backend.CallResourceHandler = (*AppEngineDatasource)(nil)
+	// We're not interested in streaming
+	// _ backend.StreamHandler         = (*SampleDatasource)(nil)
+	_ instancemgmt.InstanceDisposer = (*AppEngineDatasource)(nil)
+)
+
+type appEngineDataSourceSourceSettings struct {
+	ApiUrl string `json:"apiUrl"`
+	Realm  string `json:"realm"`
+	Token  string `json:"token"`
+}
+
+func newAppEngineDatasourceSettings(instanceSettings backend.DataSourceInstanceSettings) (appEngineDataSourceSourceSettings, error) {
+	var settings appEngineDataSourceSourceSettings
+	if err := json.Unmarshal(instanceSettings.JSONData, &settings); err != nil {
+		return appEngineDataSourceSourceSettings{}, err
+	}
+	return settings, nil
+}
+
+// NewAppEngineDatasource creates a new datasource instance.
+func NewAppEngineDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	log.DefaultLogger.Info("NewAppEngineDatasource called with", "backend_settings", settings)
+
+	datasource := &AppEngineDatasource{}
+	dsSettings, err := newAppEngineDatasourceSettings(settings)
+	if err != nil {
+		log.DefaultLogger.Error("Cannot read settings", "error", err)
+		return nil, err
+	}
+
+	// If localhost is used, one must specify AppEngine individual URL and no SSL
+	astarteAPIClient, err := client.New(
+		client.WithBaseURL(dsSettings.ApiUrl),
+		// client.WithHTTPClient(&http.Client{
+		// 	Timeout: time.Second * 30,
+		// 	Transport: &http.Transport{
+		// 		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		// 	},
+		// }),
+		// client.WithAppengineURL("http://localhost:4002"),
```

Comment:
I think these commented lines are not needed

Reply:
That's right, leftover